bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -551,16 +551,27 @@ private static boolean isConvertibleToArrayType(Object sourceValue, BArrayType t
         }
         ArrayValue source = (ArrayValue) sourceValue;
         Type targetTypeElementType = targetType.getElementType();
-        if (source.getType().getTag() == TypeTags.ARRAY_TAG) {
-            Type sourceElementType = ((BArrayType) source.getType()).getElementType();
+        Type sourceType = source.getType();
+        if (sourceType.getTag() == TypeTags.ARRAY_TAG) {
+            Type sourceElementType = ((BArrayType) sourceType).getElementType();
             if (isNumericType(sourceElementType) && isNumericType(targetTypeElementType)) {
                 return true;
             }
         }
+        int targetSize = targetType.getSize();
+        long sourceSize = source.getLength();
+        boolean returnVal = true;
+        if (!TypeChecker.hasFillerValue(targetType) && sourceSize < targetSize) {
+            addErrorMessage(0, errors, "array cannot be expanded to size '" + targetSize + "' because, the target " +
+                    "type '" + targetType + "' does not have a filler value");
+            returnVal = false;
+            if (errors.size() >= MAX_CONVERSION_ERROR_COUNT + 1) {
+                return false;
+            }
+        }
```

Comment:
```suggestion
        if (!TypeChecker.hasFillerValue(targetType) && sourceSize < targetSize) {
            addErrorMessage(0, errors, "array cannot be expanded to size '" + targetSize + "' because, the target " +
                    "type '" + targetType + "' does not have a filler value");
            return false;
        }
```
I think we can return here because going through each element and checking whether is it convertible is redundant here.